front/src/common/RollingStockContext.tsx

```diff
@@ -0,0 +1,41 @@
+import { createContext, useContext, useMemo } from 'react';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { LightRollingStockWithLiveries } from 'common/api/osrdEditoastApi';
+
+type RollingStockContextValue = {
+  rollingStockMap: Map<string, LightRollingStockWithLiveries>;
+  isError: boolean;
+  rollingStocks: LightRollingStockWithLiveries[] | null;
+};
+
+const RollingStockContext = createContext<RollingStockContextValue | null>(null);
+
+export const useRollingStockContext = (): RollingStockContextValue => {
+  const context = useContext(RollingStockContext);
+  if (!context) {
+    throw new Error('useRollingStockContext must be used within a RollingStockContextProvider');
+  }
+  return context;
+};
+
+type RollingStockContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const RollingStockContextProvider = ({ children }: RollingStockContextProviderProps) => {
+  const { data, isError } = osrdEditoastApi.endpoints.getLightRollingStock.useQuery({
+    pageSize: 1000,
+  });
+
+  const rollingStocks = data?.results ?? [];
```

Comment:
If `data.results` isn't set, then we create a new array each time the component renders. This can potentially lead to a lot of unnecessary work: all `useEffect()` and `useMemo()` depending on this value are re-triggered at each re-render.

We can avoid this by moving the `rollingStocks` variable into the `useMemo()` below.